presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -744,27 +748,27 @@ private AmazonS3 createAmazonS3Client(Configuration hadoopConfig, ClientConfigur
         }
     }
 
-    private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configuration conf)
+    private Optional<AWSCredentialsProvider> createAwsCredentialsProvider(URI uri, Configuration conf)
     {
         Optional<AWSCredentials> credentials = getAwsCredentials(uri, conf);
         if (credentials.isPresent()) {
-            return new AWSStaticCredentialsProvider(credentials.get());
+            return Optional.of(new AWSStaticCredentialsProvider(credentials.get()));
         }
 
         if (useInstanceCredentials) {
-            return InstanceProfileCredentialsProvider.getInstance();
+            return Optional.of(InstanceProfileCredentialsProvider.getInstance());
         }
 
         if (iamRole != null) {
-            return new STSAssumeRoleSessionCredentialsProvider.Builder(this.iamRole, "presto-session").build();
+            return Optional.of(new STSAssumeRoleSessionCredentialsProvider.Builder(this.iamRole, "presto-session").build());
         }
 
         String providerClass = conf.get(S3_CREDENTIALS_PROVIDER);
         if (!isNullOrEmpty(providerClass)) {
-            return getCustomAWSCredentialsProvider(uri, conf, providerClass);
+            return Optional.of(getCustomAWSCredentialsProvider(uri, conf, providerClass));
         }
 
-        throw new RuntimeException("S3 credentials not configured");
+        return Optional.empty();
```

Comment:
Why not `com.amazonaws.auth.DefaultAWSCredentialsProviderChain#getInstance`?
Then you perhaps wouldn't need the other changes, like making it Optional.

Reply:
I did this to be consistent with the Glue client, but can make this change.

Reply:
Agreed. This is what the client does internally:
```java
private AWSCredentialsProvider resolveCredentials() {
    return (credentials == null) ? DefaultAWSCredentialsProviderChain.getInstance() : credentials;
}
```
Not setting `credentials` results in it being null.

Reply:
Thanks, I will get rid of the Optional.